src/hooks/useCountdown.ts

```diff
@@ -0,0 +1,49 @@
+import { useState, useEffect } from "react"
+import {
+  dateToUnixTimestamp,
+  ONE_SEC_IN_MILISECONDS,
+  unixTimestampToTimeUnits,
+  addLeadingZero,
+  TimeUnits,
+} from "../utils"
+
+const useCountdown = (
+  targetDateInUnix: number,
+  addLeadingZeroes?: boolean,
+  onComplete?: (targetDateInUnix: number) => void
+): TimeUnits => {
+  const [diff, setDiff] = useState(targetDateInUnix - dateToUnixTimestamp())
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const diff = targetDateInUnix - dateToUnixTimestamp()
+      if (diff === 0) {
+        if (onComplete) {
+          onComplete(targetDateInUnix)
+        }
+        clearInterval(interval)
+      }
+
+      setDiff(diff)
+    }, ONE_SEC_IN_MILISECONDS)
+
+    return () => clearInterval(interval)
+  }, [targetDateInUnix])
```

Comment:
Missing `onComplete` prop in the dependency array.

Reply:
Fixed in 597affe086e05227d319412e2e387d4fc1f6976a